junit-jupiter-engine/src/main/java/org/junit/jupiter/engine/descriptor/ClassTestDescriptor.java

```diff
@@ -114,10 +115,13 @@ public final boolean isContainer() {
 		return true;
 	}
 
+	@Override
+	public LegacyReportingInfo getLegacyReportingInfo() {
+		return new DefaultLegacyReportingInfo(null, this.getDisplayName());
+	}
+
 	private static String generateDefaultDisplayName(Class<?> testClass) {
-		String name = testClass.getName();
-		int index = name.lastIndexOf('.');
-		return name.substring(index + 1);
+		return testClass.getName();
```

Comment:
I was indeed dubious about this (also in the open questions), but some tests for example: https://github.com/junit-team/junit5/blob/master/junit-platform-surefire-provider/src/test/java/org/junit/platform/surefire/provider/RunListenerAdapterTests.java#L70

expect the name with the package prefix 
org.junit.platform.surefire.provider.RunListenerAdapterTests$MyTestClass

and not:RunListenerAdapterTests$MyTestClass.

so what would be the clean way to fix this?